blog/2019-03-29-why-haskell.md

```diff
@@ -0,0 +1,108 @@
+
+### Ability to pivot
+
+> A modest test suite + the compiler gives you the ability to refactor pretty much any part of your application without fear
+
+Start-ups are primarily economic experiments and don’t necessarily place the concerns of its developers first. We all want to write great software that is elegant, has high test coverage and goes through many rounds of code reviews before it gets committed. In early stages of a product, this may not always be possible as there will be time pressure from external clients, and an understanding that what we build today might be thrown out tomorrow due to scope changes or a pivot.
+
+Haskell helps us maintain software quality in such scenarios, by forcing the implicit assumptions that we make in our head to be explicitly spelled out. The presence of property testing libraries means that we can codify our assumptions about the program and have it do the hard work of verifying if what we wrote aligns to our assumptions.
+
+This also means that code for a feature that we wrote many months ago and haven’t touched since can be expected to continue to work and be worked on at some point in the future.
+
+### Cost efficiency
+
+> If you need to scale, Haskell’s runtime efficiency can save you a substantial amount of money, especially at the beginning when you are the most resource constrained.
+
+The Haskell backend that replaced our old backend was significantly more efficient. It allowed us to run fewer servers to support the same workload. The cost savings from a smaller AWS bill can make a difference if your startup is in the phase where every dollar counts.
+
+Here is a chart of our AWS expenditure as we switched to a Haskell backend:
+
+<Insert chart>
+
+## The Bad
+
+### Lack of conventions
+
+> There are very few well-trodden paths in Haskell, expect to make lots of decisions about your stack.
+
+Apart from things like `Yesod`, there aren’t really many framework style Haskell projects that come with best practices on how to structure your application or guides on how to do common tasks like handling file uploads, user authentication, database management etc.
+
+Working on Haskell projects feels very much like working with `react` projects in that you have to bring in external libraries piecemeal for most of the features you want and the libraries aren’t necessarily designed with the assumption that they will be used together in that particular combination.
+
+Conventions also codify some hard-learned lessons, in their absence most of those lessons have to be re-learned by your team.
+
+### Lack of libraries
+
+> While the quality of libraries is excellent, the quantity can be rather limiting. Don’t expect the equivalent of passportjs or paperclip.
```

Comment:
Do you want to make all libraries/frameworks/tools mentioned in the post as links? That could be handy I guess